src/SprykerEco/Zed/Computop/Communication/ComputopCommunicationFactory.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Computop\Communication;
+
+use Generated\Shared\Transfer\MessageTransfer;
+use Spryker\Zed\Kernel\Communication\AbstractCommunicationFactory;
+use SprykerEco\Zed\Computop\ComputopDependencyProvider;
+
+/**
+ * @method \SprykerEco\Zed\Computop\ComputopConfig getConfig()
+ * @method \SprykerEco\Zed\Computop\Persistence\ComputopQueryContainerInterface getQueryContainer()
+ */
+class ComputopCommunicationFactory extends AbstractCommunicationFactory
+{
+
+    /**
+     * @return \SprykerEco\Zed\Computop\Dependency\Facade\ComputopToSalesInterface
+     */
+    public function getSalesFacade()
+    {
+        return $this->getProvidedDependency(
+            ComputopDependencyProvider::FACADE_SALES
+        );
+    }
+
+    /**
+     * @param int $orderId
+     *
+     * @return \Orm\Zed\Computop\Persistence\SpyPaymentComputop mixed
+     */
+    public function getComputopPaymentByOrderId($orderId)
+    {
+        return $this->getQueryContainer()->queryPaymentByOrderId($orderId)->findOne();
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Computop\Dependency\Facade\ComputopToCalculationInterface
+     */
+    public function getCalculationFacade()
+    {
+        return $this->getProvidedDependency(
+            ComputopDependencyProvider::FACADE_CALCULATION
+        );
+    }
+
+    /**
+     * @return \Spryker\Zed\Messenger\Business\MessengerFacadeInterface
+     */
+    public function getFlashMessengerFacade()
+    {
+        return $this->getProvidedDependency(
+            ComputopDependencyProvider::FACADE_FLASH_MESSENGER
+        );
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\MessageTransfer
+     */
+    public function createMessage()
```

Comment:
createMessageTransfer maybe? But this method is not even needed, you can directly new the transfer where needed, no factory method required.